meerk40t/gui/simulation.py

```diff
@@ -461,63 +589,301 @@ def on_mouse_right_down(self, event=None):
-        self.combo_device.SetItems(spools)
-        self.combo_device.SetSelection(index)
-        self.on_combo_device(None)
+    # @signal_listener("activate;device")
+    # def on_activate_device(self, origin, device):
+    #     self.available_devices = self.context.kernel.services("device")
+    #     self.selected_device = self.context.device
+    #     spools = []
+    #     index = -1
+    #     for i, s in enumerate(self.available_devices):
+    #         if s is self.selected_device:
+    #             index = i
+    #             break
+    #     spools = [s.label for s in self.available_devices]
+    #     self.combo_device.Clear()
+    #     self.combo_device.SetItems(spools)
+    #     self.combo_device.SetSelection(index)
+    #     self.on_combo_device(None)
 
     @signal_listener("plan")
     def on_plan_change(self, origin, plan_name, status):
         if plan_name == self.plan_name:
             self._refresh_simulated_plan()
 
+    def on_radio_playback_mode(self, event):
+        self._playback_cuts = self.radio_cut.GetValue()
+        default = 0
+        if self.radio_cut.GetValue():
+            default = 0
+        elif self.radio_time_seconds.GetValue():
+            default = 1
+        elif self.radio_time_minutes.GetValue():
+            default = 2
+        self.context.simulation_mode = default
+        self._set_slider_dimensions()
+
+    def on_listbox_operation_rightclick(self, event):
+        def remove_operation(event):
+            idx = self.list_operations.GetSelection()
+            self.cutplan.plan.pop(idx)
+            self.context.signal("plan", self.plan_name, 1)
+            # self._refresh_simulated_plan()
+
+        def append_operation(operation):
+            def check(event):
+                self.cutplan.plan.append(my_operation)
+                self.context.signal("plan", self.plan_name, 1)
+
+            my_operation = operation
+            return check
+
+        def insert_operation(operation):
+            def check(event):
+                self.cutplan.plan.insert(
+                    self.list_operations.GetSelection(), my_operation
+                )
+                self.context.signal("plan", self.plan_name, 1)
+
+            my_operation = operation
+            return check
+
+        if self.list_operations.GetSelection() < 0:
+            return
+
+        gui = self
+
+        menu = wx.Menu()
+        self.Bind(
+            wx.EVT_MENU,
+            remove_operation,
+            menu.Append(
+                wx.ID_ANY,
+                _("Remove operation"),
+                _("Removes the current operation from the active cutplan"),
+            ),
+        )
+        standards = (
+            ("Home", "util home", ""),
+            ("Goto Origin", "util goto", "0,0"),
+            ("Beep", "util console", "beep"),
+            ("Interrupt", "util console", 'interrupt "Spooling was interrupted"'),
+            ("Console", "util console", ""),
```

Comment:
Console doesn't make sense to add since it'll show up blank since it has no command.